Proportional set size (PSS) better tracks shared memory usage.

jrafanie opened the pull request:
If you fork processes, there can be a significant amount of memory that is shared with
multiple processes. Resident set size (RSS) treats 1k of shared and 1k of private memory as
consuming the same amount of memory.  Using RSS to describe how much memory each
process is using is not accurately accounting for the shared memory being
distributed between multiple processes.

From: https://github.com/djberg96/sys-proctable/pull/49

```
PSS is the size of private pages added to each shared mapping's size
divided by the number of processes that share it. It is meant to provide a
better representation of the amount of memory actually used by a process.

If a process has 4k of private pages, 4k of shared pages shared with one
other process, and 3k of pages shared with two other processes, the PSS
is:

4k + (4k / 2) + (3k / 3) = 7k
```

In the above example, RSS would report:
```
4k + 4k + 3k = 11k
```

jrafanie commented on gems/pending/Gemfile:
```diff
@@ -46,9 +46,7 @@ gem "sys-uname",               "~>1.0.1",           :require => false
-if RbConfig::CONFIG["host_os"].include?("darwin") # This is only used on Mac
-  gem 'sys-proctable',  "~> 0.9",      :require => false
-end
+gem 'sys-proctable',  :github => "djberg96/sys-proctable", :branch => :master
```
We'll go back to the next release once this is tested and released.

jrafanie replied:
@djberg96 I tested this and it looks good!  We can release a new gem and I'll change this to use that version.  Thanks!

jrafanie commented:
Here's what I have so far cc @djberg96 @Fryguy

Fryguy commented on gems/pending/util/miq-process.rb:
```diff
@@ -111,6 +112,7 @@ def self.processInfo(pid = nil)
       cpu_total /= MiqSystem.num_cpus
       percent_cpu             = (1.0 * result[:cpu_time]) / cpu_total
       result[:percent_cpu]    = round_to(percent_cpu * 100.0, 2)
+      result[:proportional_set_size] = Sys::ProcTable.ps(Process.pid).smaps.pss
```
Might as well do this on Mac, too, right?

jrafanie replied:
Yeah, good luck figuring out how to get that information.

jrafanie replied:
:trollface:  @bdunne ^

Fryguy commented on gems/pending/util/miq-process.rb:
```diff
@@ -1,6 +1,7 @@
 # encoding: US-ASCII
 
 require 'sys-uname'
+require 'sys-proctable'
```
Should this be a slash or a dash?  The line you deleted below was a slash.

djberg96 replied:
@Fryguy It works either way. :)

djberg96 commented:
:+1:

jrafanie commented on gems/pending/util/miq-process.rb:
```diff
@@ -1,6 +1,7 @@
 # encoding: US-ASCII
 
 require 'sys-uname'
+require 'sys/proctable'
```
@Fryguy changed this to use `sys/proctable` instead of `sys-proctable` for consistency in this PR.

jrafanie commented:
@miq-bot remove_label wip

jrafanie commented:
@Fryguy @djberg96 this one is ready for review/merge if it passes.  I changed to use ~>1.0 of sys-proctable, added tests for the status_update methods and reversed double negative logic.

jrafanie commented:
Fixed whitespace rubocop complaint.

djberg96 commented:
LGTM. :+1:
